DigitalLearningSolutions.Web/Services/UserService.cs

```diff
@@ -411,6 +415,35 @@ public void SetCentreEmails(int userId, Dictionary<int, string?> centreEmailsByC
             }
         }
 
+        public EmailVerificationDetails? GetEmailVerificationDetails(string email, string code)
+        {
+            var primaryEmailVerificationDetails = userDataService.GetPrimaryEmailVerificationDetails(code);
+            var centreEmailVerificationDetails = userDataService.GetCentreEmailVerificationDetails(code);
+
+            if (primaryEmailVerificationDetails != null && centreEmailVerificationDetails != null)
+            {
+                throw new Exception("Did not expect multiple records to match email verification code");
+            }
+
+            if (email == primaryEmailVerificationDetails?.Email)
+            {
+                return primaryEmailVerificationDetails;
+            }
+
+            if (email == centreEmailVerificationDetails?.Email)
+            {
+                return centreEmailVerificationDetails;
+            }
+
+            return null;
+        }
+
+        public void SetEmailVerified(int userId, string email, DateTime verifiedDateTime)
+        {
+            userDataService.SetPrimaryEmailVerified(userId, verifiedDateTime);
+            userDataService.SetCentreEmailVerified(userId, email, verifiedDateTime);
+        }

```

Comment:
Shouldn't we be passing `email `to the SetPrimaryEmailVerified method as well so it only sets email to verified if the email matches the user's primary email?

Reply:
I suppose we _could_, but it's not strictly speaking necessary (for the current implementation, at least). You won't have got here without first having confirmed that an email/code resolves to a particular userId. It is necessary to pass email into `SetCentreEmailVerified` because we don't have `delegateId` available here, and need to be able to update just one row in the `UserCentreDetails` table, but this doesn't hold for the `Users` table where there is only one row per `userId`.

Reply:
Maybe I'm missing something but it seems like if a user clicks a link to verify their centre email address, it will automatically verify their primary email as well?

Reply:
You are not missing something. I was missing something. You are right, and I will change it!